src/main/java/org/springframework/data/couchbase/core/ReactiveReplaceByIdOperationSupport.java

```diff
@@ -86,13 +86,17 @@ public ReactiveReplaceByIdOperationSupport(final ReactiveCouchbaseTemplate templ
 			return Flux.fromIterable(objects).flatMap(this::one);
 		}
 
-		private ReplaceOptions buildReplaceOptions() {
+		private ReplaceOptions buildReplaceOptions(T object) {
 			final ReplaceOptions options = ReplaceOptions.replaceOptions();
 			if (persistTo != PersistTo.NONE || replicateTo != ReplicateTo.NONE) {
 				options.durability(persistTo, replicateTo);
 			} else if (durabilityLevel != DurabilityLevel.NONE) {
 				options.durability(durabilityLevel);
 			}
+			long cas = template.support().getCas(object);
+			if (cas != 0) {
```

Comment:
0 cas == not present so if 0 is passed in its fine

Reply:
fixed